### PR TITLE
i349: Create 'don't serialize if null' attribute for models

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-349

### CODE REVIEW

**Comment by MartinEden** [comment]
See also linked PR: https://github.com/vimc/montagu-webmodels/pull/3

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt [hunk 1]
```diff
@@ -14,30 +14,29 @@ class UserController(context: ControllerContext) : AbstractController(context)
...
+            val roles = allRoles.filter { roleReadingScopes.encompass(Scope.parse(it)) }
+            return user.copy(roles = roles)
+        }
+        else
+        {
+            return user
         }
-
-        return userWithRoles
     }
```
Comment (MartinEden) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt]: I also made some changes here with the goal (hopefully) of improving clarity:

* I flipped around the if statement, to make it read positive condition first
* I noticed that we only needed to know about the scopes for the `roles.read` permission, and then that led me to be able to simplify the role filter logic.

File: src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt [hunk 2]
```diff
@@ -20,7 +21,7 @@ class UserControllerTests : ControllerTests<UserController>()
     fun `getUser returns user without roles`()
     {
         val userName = "test"
-        val user = User("test", "test name", "test@test.com", null)
+        val user = User("test", "test name", "test@test.com", null, null)
```
Comment (littledice) [inline src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt]: can use new constructor here (i.e. do without the last null)

File: src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt [hunk 3]
```diff
@@ -42,27 +43,26 @@ class UserControllerTests : ControllerTests<UserController>()
     {
         val userName = "test"
 
-        val userWithRoles = UserWithRoles("test", "test name", "test@test.com", null,
-                listOf(RoleAssignment("user", null, null),
-                        RoleAssignment("member", "modelling-group", "IC-Garske")))
+        val user = User("test", "test name", "test@test.com", null, null)
+        val expectedRoles = listOf(
```
Comment (littledice) [inline src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt]: similarly here

File: src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt [hunk 4]
```diff
@@ -71,16 +71,19 @@ class UserControllerTests : ControllerTests<UserController>()
     {
         val userName = "test"
 
-        val userWithRoles = UserWithRoles("test", "test name", "test@test.com", null,
-                listOf(RoleAssignment("user", null, null),
-                        RoleAssignment("member", "modelling-group", "IC-Garske"),
-                        RoleAssignment("member", "modelling-group", "group2"),
-                        RoleAssignment("member", "foo", "IC:Garske")))
+        val user = User("test", "test name", "test@test.com", null, null)
+        val roles = listOf(
```
Comment (littledice) [inline src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt]: similarly here

File: src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt [hunk 5]
```diff
@@ -91,18 +94,18 @@ class UserControllerTests : ControllerTests<UserController>()
...
+        val actualRoles = controller.getUser(context).roles
 
         assertThat(actualRoles).hasSameElementsAs(expectedRoles)
     }
 
     @Test
     fun `getUsers returns all users without roles`()
     {
-        val users = listOf(User("test1", "Test Name 1   ", "test1@test.com", null))
+        val users = listOf(User("test1", "Test Name 1   ", "test1@test.com", null, null))
```
Comment (littledice) [inline src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt]: similarly here

**Comment by MartinEden** [comment]
Fixed all of those hopefully :)

**Pull request merged**